src/subscription_manager/repolib.py

```diff
@@ -60,13 +60,98 @@ def manage_repos_enabled():
     except configparser.Error as e:
         log.exception(e)
         return True
-
-    if manage_repos is None:
-        return True
+    else:
+        if manage_repos is None:
+            return True
 
     return bool(manage_repos)
 
 
+class YumPluginManager(object):
+    """
+    Instance of this class is used for automatic enabling of yum plugins.
+    """
+
+    YUM_PLUGIN_DIR = '/etc/yum/pluginconf.d'
+
+    # List of yum plugins in YUM_PLUGIN_DIR which are automatically enabled
+    # during sub-man CLI/GUI start
+    YUM_PLUGINS = ['subscription-manager', 'product-id']
+
+    YUM_PLUGIN_ENABLED = 1
+    YUM_PLUGIN_DISABLED = 0
+
+    @staticmethod
+    def is_auto_enable_enabled():
+        """
+        Automatic enabling of yum plugins can be explicitly disabled in /etc/rhsm/rhsm.conf
+        Try to get this configuration.
+        :return: True, when auto_enable_yum_plugins is enabled. Otherwise False is returned.
+        """
+        try:
+            auto_enable_yum_plugins = conf['rhsm'].get_int('auto_enable_yum_plugins')
+        except ValueError as err:
+            log.exception(err)
+            auto_enable_yum_plugins = True
+        except configparser.Error as err:
+            log.exception(err)
+            auto_enable_yum_plugins = True
+        else:
+            if auto_enable_yum_plugins is None:
+                auto_enable_yum_plugins = True
+        return bool(auto_enable_yum_plugins)
+
+    @staticmethod
+    def warning_message(enabled_yum_plugins):
+        message = _('The yum plugins: %s were automatically enabled for the benefit of '
+                    'Red Hat Subscription Management. If not desired, use '
+                    '"subscription-manager config --rhsm.auto_enable_yum_plugins=0" to '
+                    'block this behavior.\n') % str(tuple(enabled_yum_plugins))
+        return message
+
+    @classmethod
+    def enable_yum_plugins(cls):
+        """
+        This function tries to enable yum plugins: subscription-manager and product-id.
+        It takes no action, when automatic enabling of yum plugins is disabled in rhsm.conf.
+        :return: It returns list of enabled plugins
+        """
+
+        # When user doesn't want to automatically enable yum plugins, then return empty list
+        if cls.is_auto_enable_enabled() is False:
+            log.info('The rhsm.auto_enable_yum_plugins is disabled')
```

Comment:
Please add more context about what action is happening or not happening. For example: "The rhsm.auto_enable_yum_plugins is disabled. Skipping the enablement of yum plugins" or something like that.

Reply:
I agree.